Sources/ASTPreprocessor/EnclosingTypeAssigner.swift

```diff
@@ -0,0 +1,80 @@
+//
+//  EnclosingTypeAssigner.swift
+//  ASTPreprocessor
+//
+//  Created by Nik on 23/11/2018.
+//
+
+import AST
+import Diagnostic
+
+/// The Enclosing Type Assignment pass for the AST.
```

Comment:
More description here again? E.g. `assigns types to expressions`?

Reply:
I mean it's literally in the name

Reply:
"Enclosing type" is not understandable without knowledge of type assignment in `.` operations.